sig-security-assessments/cluster-api/self-assessment.md

```diff
@@ -0,0 +1,1099 @@
+
+Clusters managed _by_ Cluster API are called _workload clusters_, and are
+primarily what workload operators and application developers will interact with.
+
+#### Management Clusters
+
+A management cluster is a cluster that actually runs the Cluster API components,
+contains the resources representing the cluster and is reaching out to cloud
+provider services to perform infrastructure operations. We would expect the
+Cloud Admin, Platform Operator and to some extent Workload Operators to interact
+with the Management Cluster, but not typically Application Developers. The
+diagram below illustrates the relationship between management and workload
+clusters.
+
+![Schematic of relation between management cluster and workload cluster](images/mgmt-and-workload-cluster-relationship.png)
+
+**Figure 1**: Relation between management cluster and workload cluster
+
+##### Self-managed Management Cluster
+
+A self-managed management cluster is a management cluster whose lifecycle is
+being managed by Cluster API running on the management cluster itself.
+
+#### Bootstrap Cluster
+
+A bootstrap cluster is typically a Kubernetes “cluster” running on a local
+workstation - in the Cluster API book, we typically recommend the usage of kind,
+but any developer Kubernetes distribution is sufficient. This is used to spin up
+an initial cluster with Cluster API which maybe converted _into_ a management
+cluster. This is covered later. (TODO). It is expected that platform operators
+use a bootstrap cluster to provision the initial management cluster.
+
+##### Cluster API Core Controller
+
+Cluster API Core controller is a pluggable provider in Cluster API which
+provides high level management of basic primitives in Cluster API. It provides a
+blueprint of the cluster and its components like machine deployment, machineset,
+and machines which acts as reference for cloud infrastructure providers to
+create infrastructure related components.
+
+##### Cluster API Bootstrap Kubeadm Controller
+
+Cluster API Bootstrap Kubeadm Controller is a provider which helps to convert a
+machine into a kubernetes node and thus bootstrapping the whole cluster and its
+machines. It uses kubeadm for bootstrapping kubernetes.
+
+It takes care of initializing control plane nodes and bootstrapping other nodes
+as well.
+
+Once all the nodes are bootstrapped, the Bootstrap kubeadm Controller then
+attaches the control plane and worker nodes to the cluster with their specific
+roles within the cluster.
+
+##### Cluster API Kubeadm Control Plane Controller
+
+Cluster API Kubeadm Control plane controller provides ability to manage a
+control plane, including bootstrapping, scaling and deletion of control plane
+without disturbing the whole cluster quorum.
+
+The general expectation from the control plane controller is to make sure that
+the control plane node is created with the services like etcd, kubernetes API
+server, kubernetes controller manager, scheduler, cloud controller manager,
```

Comment:
nit: cloud controller manager is not managed by KCP (so at least KCP doesn't fulfill this expectation)

Reply:
done